reservations/migrations/0073_affectingtimespan.py

```diff
@@ -0,0 +1,121 @@
+# Generated by Django 5.0.6 on 2024-06-20 16:10
+
+import django.contrib.postgres.fields
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("reservations", "0072_alter_reservation_begin_alter_reservation_end"),
+    ]
+
+    operations = [
+        # Create materialized view
+        migrations.RunSQL(
+            sql=(
+                """
+                CREATE MATERIALIZED VIEW affecting_time_spans AS
+                    SELECT
+                        res.reservation_id,
+                        array_agg(res.ru_id ORDER BY res.ru_id) AS affected_reservation_unit_ids,
+                        res.buffered_start_datetime,
+                        res.buffered_end_datetime,
+                        res.buffer_time_before,
+                        res.buffer_time_after,
+                        res.is_blocking
+                    FROM (
+                        SELECT DISTINCT
+                            r.id as reservation_id,
+                            unnest(ruh.related_reservation_unit_ids) as ru_id,
+                            (r.begin - r.buffer_time_before) as buffered_start_datetime,
+                            (r.end + r.buffer_time_after) as buffered_end_datetime,
+                            r.buffer_time_before as buffer_time_before,
+                            r.buffer_time_after as buffer_time_after,
+                            (CASE WHEN UPPER(r."type") = 'BLOCKED' THEN true ELSE false END) as is_blocking
+                        FROM reservation r
+                        INNER JOIN "reservation_reservation_unit" rru ON r.id = rru.reservation_id
+                        INNER JOIN "reservation_unit_hierarchy" ruh ON rru.reservationunit_id = ruh.reservation_unit_id
+                        WHERE (
+                            (r.end + r.buffer_time_after)::date >= NOW()::date
+                            AND UPPER(r.state) IN ('CREATED', 'CONFIRMED', 'WAITING_FOR_PAYMENT', 'REQUIRES_HANDLING')
+                        )
+                    ) res
+                    GROUP BY
+                        res.reservation_id,
+                        res.buffered_start_datetime,
+                        res.buffered_end_datetime,
+                        res.buffer_time_before,
+                        res.buffer_time_after,
+                        res.is_blocking
+                    ORDER BY res.buffered_start_datetime, res.reservation_id;
```

Comment:
Does it matter [that](https://www.postgresql.org/docs/current/sql-refreshmaterializedview.html):
> If there is an ORDER BY clause in the materialized view's defining query, the original contents of the materialized view will be ordered that way; but REFRESH MATERIALIZED VIEW does not guarantee to preserve that ordering.

Not because of `AffectingTimeSpan.Meta.ordering`?

Reply:
Oh, good to know. Still, I think this should be fine since AffectingTimeSpans are ordered chronologically when they are merged in `FirstReservableTimeHelper._get_elements_for_reservation_units`.